connexion/decorators/response.py

```diff
@@ -52,9 +52,9 @@ def validate_response(self, data, status_code, headers, url):
                 raise NonConformingResponseBody(message=str(e))
 
         if response_definition and response_definition.get("headers"):
-            response_definition_header_keys = response_definition.get("headers").keys()
-            header_keys = headers.keys()
-            missing_keys = response_definition_header_keys - header_keys
+            required_header_keys = {k for (k, v) in response_definition.get("headers").items()
+                                    if v.get('required', False)}
+            missing_keys = required_header_keys - headers.keys()
```

Comment:
Seems like the tests are failing because of this line: apparently, `headers` (in the case of Flask) is actually a `werkzeug.datastructures.Headers` object instead of a regular Python `dict`. 😞 

Guess we'll have to change it back to the "python2.7-style" with sets:
```suggestion
            missing_keys = required_header_keys - set(headers.keys())
```

Reply:
Applied the change